From f9a65e5cb4616159bd3ec1c90ed565a1a8920f91 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Thu, 29 Apr 2021 14:56:32 +0000
Subject: [PATCH] imap: fix build with boost-1.76.0

* explicitly include boost/scoped_array.hpp, because it's no longer implicitly
  included through:
  mojomail-imap/2.0.0-99+gitAUTOINC+446c53ce81-r0/git/imap/src/protocol/FetchResponseParser.cpp:20
  mojomail-imap/2.0.0-99+gitAUTOINC+446c53ce81-r0/recipe-sysroot/usr/include/boost/regex.hpp:31
  mojomail-imap/2.0.0-99+gitAUTOINC+446c53ce81-r0/recipe-sysroot/usr/include/boost/regex/v4/regex.hpp:32
  mojomail-imap/2.0.0-99+gitAUTOINC+446c53ce81-r0/recipe-sysroot/usr/include/boost/regex/v4/regex_workaround.hpp:43
  mojomail-imap/2.0.0-99+gitAUTOINC+446c53ce81-r0/recipe-sysroot/usr/include/boost/scoped_array.hpp

  because since:
  https://github.com/boostorg/regex/commit/72da1fdc298dbe8196a32d82251bb31faf715da6#diff-d0a87a488e710b6d9b77dee194418149f06fa58bb7250f1872d1ab286b494dc1
  regex/v5/regex.hpp is included by default and that doesn't include scoped_array.

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 imap/src/protocol/FetchResponseParser.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/imap/src/protocol/FetchResponseParser.cpp b/imap/src/protocol/FetchResponseParser.cpp
index 0c184b8..14719b2 100644
--- a/imap/src/protocol/FetchResponseParser.cpp
+++ b/imap/src/protocol/FetchResponseParser.cpp
@@ -18,6 +18,7 @@
 
 #include "protocol/FetchResponseParser.h"
 #include <boost/regex.hpp>
+#include <boost/scoped_array.hpp>
 #include "parser/ImapParser.h"
 #include "parser/Rfc3501Tokenizer.h"
 #include "parser/SemanticActions.h"
